src/book/03-guides/04-general/04-visualmesh.mdx

```diff
@@ -0,0 +1,205 @@
+
+- `docker` is the call to docker
+- `build` is the main command being sent to docker
+- `.` is telling docker that the build context is the current directory. Docker will look for a file named `Dockerfile` in the build context telling it what to build (so in this case, the current directory).
+- `--pull` tells docker to always attempt to pull a newer version of the image
+- `-t visualmesh:latest` tells docker that we want to tag the built image as `visualmesh:latest`. We'll see later that when we tell docker to run a container, that we tell docker which image to use by referencing this tag.
+
+</details>
+
+## Make a dummy dataset
+
+For the sake of this guide, we'll use a dummy dataset containing the same image repeated over and over again. _While this will lead to extreme overfitting, it's an easy way to quickly check that the code is running correctly._ Download the example image, mask, and lens file [dummy_data.zip](assets/dummy_data.zip) to your downloads folder.
+
+Now, we're going to make a dummy dataset by repeating these files. We can do this using a bash loop. Assuming that we have the dummy data in `~/Downloads/dummy_data`, and we're going to put the repeated data in `~/Downloads/repeated_data`, run the four commands below:
+
+```bash
+cd ~/Downloads
+
+unzip dummy_data.zip
+
+mkdir repeated_data
+
+for i in {1..1000}; do cp ./dummy_data/image.jpg ./repeated_data/image${i}.jpg && cp ./dummy_data/mask.png ./repeated_data/mask${i}.png && cp ./dummy_data/lens.yaml ./repeated_data/lens${i}.yaml; done
+```
+
+To ensure that the data was copied sucessfully run the command below which will print all the filenames to the terminal window. If you would like to clean up your display, you can use the `clear` command.
+
+```bash
+ls ~/Downloads/repeated_data
+clear
+```
+
+### Install the prerequisites to make a tfrecord
+
+We want to convert our dataset from image and text files to the tfrecord format that the visual mesh expects, but to do this we need to install some python libraries to allow python to work with tfrecords. Feel free to use something like conda or virtualenv if you're concerned about ending up with conflicting library versions.
+
+```bash
+pip3 install tensorflow tqdm
+```
+
+### Make a set of tfrecord
+
+Having made the repeated dataset and installed the libraries required to work with tfrecords, we can now put it into the tfrecord format the visual mesh wants using the `make_dataset.py` script. By default, the visual mesh will look for the dataset inside a folder named `dataset`, so we'll navigate back to the visual mesh repo, make the `dataset` folder, then make the `.tfrecord` files required using the `make_dataset.py` script:
+
+```bash
+cd ~/code/VisualMesh
+mkdir dataset
+./training/make_dataset.py ~/Downloads/repeated_data ./dataset
+```
+
+<Alert type='info'>The `make_dataset.py` won't complain if you give it an empty folder, so make sure that the created tfrecord files are not empty.</Alert>
```

Comment:
A weird quirk of MDX: it won't compile the tag's content as Markdown if it's not delimited by new lines:

```suggestion
<Alert type='info'>

The `make_dataset.py` won't complain if you give it an empty folder, so make sure that the created tfrecord files are not empty.

</Alert>
```